### PR TITLE
Add basic vector search and RAG-based search functionality

### PR DESCRIPTION
**Pull request by demid-ns**
Changes:
- Updated product ingestion process to generate a vector for each product
- Added an endpoint for basic vector search functionality
- Added an endpoint for RAG-based search to enhance search results
- Cleaned up project structure

### CODE REVIEW

File: src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Text.Json;
+using KernelMemory.Ecommerce.Sample.Api.Application.CsvReader;
+using KernelMemory.Ecommerce.Sample.Api.Domain;
+using Microsoft.KernelMemory;
+
+namespace KernelMemory.Ecommerce.Sample.Api.Application.ProductIngestionCommand;
+
+public sealed record ProductIngestionCommand(Stream ProductsFileStream) : ICommand;
+
+public sealed class IngestProductsCommandHandler(
```
Comment (denbell5) [inline src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs:10]: @demid-ns rename to ProductIngestionCommandHandler?

File: src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+        // Note: The GenerateEmbeddingsParallelHandler could potentially be used here as an alternative.
+        // However, it is currently experimental and may not yet be stable for production use
+        var importTasks = readingResult.Value.Select(product =>
+            memory.ImportTextAsync(
+                JsonSerializer.Serialize(product),
+                documentId: product.Id.ToString(),
+                cancellationToken: cancellationToken)).ToArray();
+
+        await Task.WhenAll(importTasks);
```
Comment (denbell5) [inline src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs:29]: @demid-ns is it ok to not have degree of parallelism? I don't know how many importTasks can be there protentially so probablt it's ok

**Review by denbell5 — CHANGES_REQUESTED** [review]
left single comments

File: src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs (hunk 1 above)
Comment (demid-ns) [reply src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs]: Good catch! Fixed

File: src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs (hunk 2 above)
Comment (demid-ns) [reply src/KernelMemory.Ecommerce.Sample.Api/Application/ProductIngestionCommand/ProductIngestionCommand.cs:29]: I think this is OK for the current stage. I've created an [issue](https://github.com/demid-ns/kernel-memory-ecommerce-sample/issues/2) to investigate `GenerateEmbeddingsParallelHandler` usage

**Review by denbell5 — APPROVED** [review]

**Pull request merged**